### PR TITLE
Support agent reset command

### PR DESCRIPTION
**Pull request by ascrutae**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

### CODE REVIEW

File: apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/CommandSerializer.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+public class CommandSerializer {
+
+    public final static Map<String, Deserializable> COMMAND_SERIALIZE_MAPPING = new HashMap<String, Deserializable>(5);
+
+    static {
+        COMMAND_SERIALIZE_MAPPING.put("EndpointMetadataReset", new EndpointResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("InstanceMetadataReset", new InstanceResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("NetworkAddressMetadataReset", new NetworkResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("ServiceMetadataReset", new ServiceResetCommand("FOR_DESERIALIZE"));
+        COMMAND_SERIALIZE_MAPPING.put("TraceIgnore", new TraceIgnoreCommand("FOR_DESERIALIZE"));
```
Comment (wu-sheng) [inline apm-protocol/apm-network/src/main/java/org/apache/skywalking/apm/network/trace/component/command/CommandSerializer.java]: I am totally confused about these. All command construct arguments say `serialNumber`. What are you trying to do here?

I don't think you should use this construct. @peng-yongsheng suggestion?

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/commands/CommandService.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    try {
+                        BaseCommand command = commands.take();
+
+                        if (!executeAtFirstTime(command.getSerialNumber())) {
+                            continue;
+                        }
+
+                        try {
+                            CommandExecutors.newCommandExecutor(command).execute();
+                            serialNumberCache.add(command.getSerialNumber());
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/commands/CommandService.java]: You have no debug log for command execution, which makes the command execute result unclear.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java [hunk 3]
```diff
@@ -460,6 +461,11 @@ private void finish() {
                 finishedSegment.setIgnore(true);
             }
         }
+
+        if (segment.createTime() < RemoteDownstreamConfig.Agent.INSTANCE_REGISTED_TIME) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java]: You need to create a method for this. This is also required when inject/extract and segment send to oap-server.

File: apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/commands/executor/InstanceResetCommandExecutor.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    public InstanceResetCommandExecutor(InstanceResetCommand baseCommand) {
+        super(baseCommand);
+    }
+
+    @Override
+    public void execute() {
+        RemoteDownstreamConfig.Agent.SERVICE_ID = DictionaryUtil.nullValue();
+        RemoteDownstreamConfig.Agent.SERVICE_INSTANCE_ID = DictionaryUtil.nullValue();
+        RemoteDownstreamConfig.Agent.INSTANCE_REGISTED_TIME = DictionaryUtil.nullValue();
```
Comment (wu-sheng) [inline apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/commands/executor/InstanceResetCommandExecutor.java]: Why don't you clear other caches?

**Comment by asfgit** [comment]
FAILURE

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/skywalking-e2e/17/

**Comment by wu-sheng** [comment]
Relaced by https://github.com/apache/skywalking/issues/3072

**Comment by wu-sheng** [comment]
-

**Pull request closed without merging**